src/main/java/nl/enjarai/doabarrelroll/compat/yacl/YACLImplementation.java

```diff
@@ -0,0 +1,158 @@
+                                        .build())
+                                .option(getBooleanOption("controls", "momentum_based_mouse", true)
+                                        .binding(false, () -> ModConfig.INSTANCE.getMomentumBasedMouse(), value -> ModConfig.INSTANCE.setMomentumBasedMouse(value))
+                                        .build())
+                                .option(getOption(ActivationBehaviour.class, "controls", "activation_behaviour", true)
+                                        .controller(EnumController::new)
+                                        .binding(ActivationBehaviour.VANILLA, () -> ModConfig.INSTANCE.getActivationBehaviour(), value -> ModConfig.INSTANCE.setActivationBehaviour(value))
+                                        .build())
+                                .build())
+                        .group(OptionGroup.createBuilder()
+                                .name(getText("hud"))
+                                .option(getBooleanOption("hud", "show_horizon", true)
+                                        .binding(false, () -> ModConfig.INSTANCE.getShowHorizon(), value -> ModConfig.INSTANCE.setShowHorizon(value))
+                                        .build())
+                                .option(getBooleanOption("controls", "show_momentum_widget", true)
+                                        .binding(true, () -> ModConfig.INSTANCE.getShowMomentumWidget(), value -> ModConfig.INSTANCE.setShowMomentumWidget(value))
+                                        .build())
+                                .build())
+                        .group(OptionGroup.createBuilder()
+                                .name(getText("banking"))
+                                .option(getBooleanOption("banking", "enable_banking", true)
+                                        .binding(true, () -> ModConfig.INSTANCE.getEnableBanking(), value -> ModConfig.INSTANCE.setEnableBanking(value))
+                                        .build())
+                                .option(getOption(Double.class, "banking", "banking_strength", false)
+                                        .controller(option -> new DoubleSliderController(option, 0.0, 100.0, 1.0))
+                                        .binding(20.0, () -> ModConfig.INSTANCE.getBankingStrength(), value -> ModConfig.INSTANCE.setBankingStrength(value))
+                                        .build())
+                                .build())
+                        .group(OptionGroup.createBuilder()
+                                .name(getText("thrust"))
+                                .collapsed(true)
+                                .option(getBooleanOption("thrust", "enable_thrust", true)
+                                        .binding(false, () -> ModConfig.INSTANCE.getEnableThrust(), value -> ModConfig.INSTANCE.setEnableThrust(value))
```

Comment:
Actually, you shouldn't be using the getters and setters for the config itself. They take other things into account, like if the server allows it or not. In this case, that could fuck with the config.